index.js

```diff
@@ -11,8 +9,50 @@ const randomstring = require('randomstring');
 const requestretry = require('requestretry');
 const tinytim = require('tinytim');
 const yaml = require('js-yaml');
+const _ = require('lodash');
+
+const MAXATTEMPTS = 5;
+const RETRYDELAY = 3000;
 const CPU_RESOURCE = 'beta.screwdriver.cd/cpu';
 const RAM_RESOURCE = 'beta.screwdriver.cd/ram';
+const TOLERATIONS_PATH = 'spec.tolerations';
+const AFFINITY_NODE_SELECTOR_PATH = 'spec.affinity.nodeAffinity.' +
+    'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms';
+
+/**
+ * Parses nodeSelector config and update intended nodeSelector in tolerations
+ * and nodeAffinity.
+ * @method setNodeSelector
+ * @param {Object} podConfig k8s pod config
+ * @param {Object} nodeSelectors key-value pairs of node selectors
+ */
+function setNodeSelector(podConfig, nodeSelectors) {
+    if (!nodeSelectors || typeof nodeSelectors !== 'object') {
+        return;
+    }
+
+    const tolerations = _.get(podConfig, TOLERATIONS_PATH, []);
+    const nodeAffinitySelectors = _.get(podConfig, AFFINITY_NODE_SELECTOR_PATH, []);
+
+    Object.keys(nodeSelectors).forEach((key) => {
+        tolerations.push({
+            key,
+            value: nodeSelectors[key],
+            effect: 'NoSchedule',
+            operator: 'Equal'
+        });
+        nodeAffinitySelectors.push({
+            matchExpressions: [{
+                key,
```

Comment:
similar here as line 39 above

Reply:
It's object shorthand property, will expand to `{'key': key}`